ADEN-4232 Porvata - video template

fraszczakszymon opened the pull request.

Commit:
SUS-1207: Optimize Phalanx write operations

Commit:
XW-2223 | set wgMetaNamespace in GlobalTitle

Commit:
XW-2223 | Fix wgMetaNamespaces in GlobalTitle

Commit:
XW-2223 | Use single quotes

Commit:
XW-2223 | Add comment

Commit:
XW-2223 | use array shorthand

Commit:
SOC-1278 Fix issues found in Special:PhalanxStats for wiki level stats

Commit:
SUS-1278 Fixed indentation

Commit:
SUS-1278 Make the getTypeNames a little more understandable and allow for bad data

Commit:
SUS-1215 Be more forgiving of older blocks

Older blocks may not have the ps_blocker_hit column set in the phalanx_stats table. Existing code was using the null coalescing operator to check both ps_blocker_hit then ps_blocker_type.  However as this value is sometimes zero as well as null, this did not do the right thing. Also add back in an old block for stats display purposes, but hide it for block creation.

Commit:
SUS-1215 Don't create a URL if the server is not present.

The GlobalTitle will create a URL even the server name is not available, leading to a broken URL.

Also update the schema to reflect reality.

Commit:
SUS-1215 Update wfMsgExt to wfMessasge(...)

Commit:
ADEN-4182 Remove state from google IMA

drets committed:
ADEN-4182 Make IMA reusable.

Commit:
ADEN-4182 Make BFAB video ad

Commit:
ADEN-4182 Extract UAP video

fraszczakszymon committed:
ADEN-4180 Override community skins for Fan Takeover

fraszczakszymon committed:
ADEN-4180 Mix overlay instead of lighten/darken

Commit:
ADEN-4182 Extract animation module

Commit:
ADEN-4182 Register callback methods by event listener

Commit:
Merge branch 'dev' into ADEN-4182

Commit:
ADEN-4182 Use UAP video for BFAA mobile

Commit:
ADEN-4182 Initialize IMA lib once

Commit:
ADEN-4182 Fix indentation

Commit:
SEO-612: Remove OasisLight from available skin in user preferences

lukaszkonieczny committed:
WW-362: Update CNW translations

lukaszkonieczny committed:
WW-362: Update CNW Checks crowdin config file

macbre committed:
Revert "SUS-1301: fix malformed Oasis messages in Spanish"

macbre committed:
ArticleComment: lazy-load metadata

macbre committed:
ArticleComment - introduce getAllMedata method

macbre committed:
WallMessage: make doSaveMetadata use ArticleComment::getRawText() which is cached and does not overwrite existing metadata

macbre committed:
WallHelper::mapParentData does not need the third argument

macbre committed:
ArticleComment: make mMetadata a private field

fraszczakszymon committed:
ADEN-4232 Expose all IMA events

fraszczakszymon committed:
ADEN-4232 Wrap IMA object

Commit:
add a restriction check in blockWiki()

macbre committed:
ArticleComment: fix a tpyo

fraszczakszymon committed:
ADEN-4232 Porvata template

fraszczakszymon committed:
ADEN-4232 Update jsdoc

WikiaTech commented on extensions/wikia/AdEngine/js/template/bfaaDesktop.js:
```diff
@@ -2,21 +2,19 @@
 	'wikia.log',
 	'wikia.window',
 	require.optional('ext.wikia.aRecoveryEngine.recovery.tweaker')
 ], function (
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has 9 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

WikiaTech commented on extensions/wikia/AdEngine/js/template/bfaaMobile.js:
```diff
@@ -1,22 +1,18 @@
 	'wikia.log',
 	'wikia.window',
 	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
 ], function (
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has 8 parameters which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

WikiaTech commented:
SonarQube analysis reported 2 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 2 major

Watch the comments in this conversation to review them.

fraszczakszymon committed:
ADEN-4232 Fix template

Commit:
XW-2307 | in infobox only image files are displayed

macbre committed:
FinishCreateWikiController: fix a PHP fatal

Error: Call to a member function getArticleID() on null in extensions/wikia/CreateNewWiki/FinishCreateWikiController.class.php:113

fraszczakszymon committed:
ADEN-4232 UAP id fallback

drets commented on extensions/wikia/AdEngine/js/template/porvata.js:
```diff
@@ -0,0 +1,48 @@
+	 * @param {object} params.height - Player height
+	 * @param {string} [params.onReady] - Callback executed once player is ready
+	 * @param {string} [params.vastUrl] - Vast URL (DFP URL with page level targeting will be used if not passed)
+	 */
```
:+1: nice, could we add similar comment to bfaa(b)?

fraszczakszymon replied:
Yes, we can :trollface:

drets commented on extensions/wikia/AdEngine/js/template/porvata.js:
```diff
@@ -0,0 +1,48 @@
+	return {
+		show: show
+	};
+});
```
EOF

drets commented on extensions/wikia/AdEngine/js/video/uapVideo.js:
```diff
@@ -0,0 +1,69 @@
+				{
+					src: 'gpt',
+					pos: params.slotName,
+					uap: params.uap || uapContext.getUapId(),
```
:+1:

Commit:
ADEN-4182 CR cleanup

drets commented on extensions/wikia/AdEngine/js/template/bfaaMobile.js:
```diff
@@ -103,43 +59,24 @@ define('ext.wikia.adEngine.template.bfaaMobile', [
-
-					params.videoTriggerElement.addEventListener('click', function () {
-						video.play(showVideo, hideVideo);
+					video.addEventListener(win.google.ima.AdEvent.Type.LOADED, function () {
```
:+1: :-1: 

:+1: - for removing `onResize.bind(null, params.aspectRatio)`
:-1: - for exposing the implementation details (a fact that we use ima)

Did you think about different approaches (at least to have our own “enum”)?

fraszczakszymon replied:
It's point of those changes to expose IMA events. I wouldn't like to wrap whole library, events, etc.

drets commented on extensions/wikia/AdEngine/js/video/videoAdFactory.js:
```diff
@@ -1,44 +1,45 @@
 
-	function create(width, height, adSlot, slotParams) {
-		var vastUrl = vastUrlBuilder.build(width / height, slotParams);
+	function create(width, height, adSlot, slotParams, vastUrl, preventImaReload) {
```
What do you think about instead of introducing a new flag in `create` method separate `play` and `reload` functionality and do something meaningful with it?

fraszczakszymon replied:
Absolutely agree. Thanks :gem:

drets commented on extensions/wikia/AdEngine/js/video/googleIma.js:
```diff
@@ -36,43 +36,88 @@ define('ext.wikia.adEngine.video.googleIma', [
 
-		return prepareVideoAdContainer(adContainer.querySelector('div'));
-	}
+		Object.keys(google.ima.AdErrorEvent.Type).forEach(function (eventKey) {
```
I see kinda similar code for `google.ima.AdErrorEvent.Type` and `google.ima.AdEvent.Type`, didn't you think about abstracting it?

fraszczakszymon replied:
AdErrorEvent has adsLoader dependency and AdEvent has adsManager

drets commented:
Minor comments, lgtm in general :+1:

Commit:
XW-2307 | fixed whitespaces, using getMediaType method

Commit:
XW-2307 | mocking getMediaType method in custom width logic test

fraszczakszymon committed:
ADEN-4232 CR fixes

fraszczakszymon committed:
ADEN-4232 Add UAP jsdocs

bognix committed:
ADEN-4232 | fix video not playing on iOS 9

kvas-damian committed:
XW-2307 | Whitelist videos in Portable Infoboxes

macbre committed:
Revert "in prod lookup phalanx using the prodTesting tag"

This reverts commit a2b9cd2fbfbd29803db07110beb024d346f6ea39.

See #11942

WikiaTech commented on extensions/wikia/PhalanxII/classes/Phalanx.class.php:
```diff
@@ -186,44 +280,120 @@ public function delete() {
+			'wgDBmysql5' => true
+		] );
+
+		$db = $wrapper->wrap( function () use ( $dbType ) {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$db". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

WikiaTech commented on extensions/wikia/PhalanxII/classes/Phalanx.class.php:
```diff
@@ -117,48 +130,128 @@ public function load() {
 		$dbw->commit();
 
 		wfProfileOut( __METHOD__ );
 		return ( $action ) ? $this->data['id'] : false;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the literal "false" boolean value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

WikiaTech commented on extensions/wikia/PhalanxII/classes/Phalanx.class.php:
```diff
@@ -117,48 +130,128 @@ public function load() {
 			$dbw->update( $this->db_table, $this->mapToDB(), array( 'p_id' => $this->data['id'] ), __METHOD__ );
-			$action = 'edit';
+			$action = static::ACTION_MODIFY_FILTER;
 		};
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1116)

WikiaTech commented on extensions/wikia/PhalanxII/classes/PhalanxStatsWikiaPager.class.php:
```diff
@@ -10,20 +12,44 @@ public function __construct( $id ) {
+			[ 'parseinline', 'replaceafter' ],
+			$type, $username, $phalanxUrl, $timestamp, $statsUrl, $url
+		);
 		$html .= Html::closeElement( 'li' );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$html". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

WikiaTech commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,124 @@ function __construct( ) {
 	}
 
 	public function index() {
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

WikiaTech commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,124 @@ function __construct( ) {
 
 	public function index() {
-		wfProfileIn( __METHOD__ );
 		$this->wg->Out->setPageTitle( wfMsg( 'phalanx-stats-title' ) );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "phalanx-stats-title" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php:
```diff
@@ -12,45 +12,124 @@ function __construct( ) {
+
+		// pull these out of the array, so they don't get used in the top rows
+		$row = $data->toArray();
+		unset( $row['text'], $row['reason'], $row['comment'], $row['ip_hex'] );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "reason" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "comment" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/ArticleComments/classes/ArticleComment.class.php:
```diff
@@ -24,7 +24,7 @@ class ArticleComment {
-	public $mMetadata;
+	private $mMetadata;
 
 	private $mText = false; // parsed HTML
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

WikiaTech commented:
SonarQube analysis reported 355 issues
* ![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) 1 critical
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 99 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 173 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 82 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) [Language.php#L3639](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/languages/Language.php#L3639): Move arguments "$prefix" after arguments without default value [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1788)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php): File "ArticleComment.class.php" has 1722 lines, which is greater than 1000 authorized. Split it into smaller files. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php#L8](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L8): Class "ArticleComment" has 64 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php#L238](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L238): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php#L614](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L614): Remove the unused function parameter "$content". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php#L795](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L795): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php#L924](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L924): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php#L925](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L925): Remove this unused "$wgTitle" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php#L1074](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L1074): Reduce this "switch/case" number of lines from 32 to at most 10, for example by extracting code into function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [ArticleComment.class.php#L1106](https://github.com/Wikia/app/blob/77232ab296604aac5eab99f3bf2cbffe6f1f20fd/extensions/wikia/ArticleComments/classes/ArticleComment.class.php#L1106): Reduce this "switch/case" number of lines from 13 to at most 10, for example by extracting code into function. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1151)
